docusaurus/docs/recipes/contentful.md

```diff
@@ -0,0 +1,511 @@
+        host: 'cdn.contentful.com', // Delivery API host
+    });
+
+    if (
+        // For server-side, check wether `query` exists and has the `cms-preview` key
+        (typeof query !== 'undefined' && Object.hasOwnProperty.call(query, 'cms-preview')) ||
+        // For client-side, check wether `window` exist and has the `cms-preview` query parameter
+        (typeof window !== 'undefined' && new URLSearchParams(window.location.search).has('cms-preview'))
+    ) {
+        // In a positive case, switch the client instance to access the Preview API instead		
+        client = createClient({
+            space: process.env.CONTENTFUL_SPACE_ID,
+            accessToken: process.env.CONTENTFUL_PREVIEW_TOKEN, // Preview API Token
+            host: 'preview.contentful.com', // Preview API host
+        });
+    }
+
+    const reducer = {
+        // Example reducer
+    };
+    
+    // Add client as a thunk middleware
+    // This will give us access to the `client` object in our action functions as an argument
+    const middlewares = applyMiddleware(thunkMiddleware.withExtraArgument({ client }));
+
+    return createStore(reducer, initialState, middlewares);
+    }
+}
+
+export default buildStore;
+```
+
+Then, in your actions scripts, you can use the provided client to fetch the content that you need, like so:
+
+```js
+// usecases/actions.js
+// Create a fetch action that fetches data from Contentful
+const fetch = () => async (dispatch, getState, { client }) => {
+    dispatch({ type: 'Fetching'})
+
+    try {
+        // Fetch content that are instances of `useCase` content model, in Spanish
+        const result = await client.getEntries({
+            content_type: 'useCase',
+            locale: 'es-ES',
+        })
+
+        // The incoming object will be fairly complex, and it's recommended to build
+        // a more conveniently structured object here so that other scrips in the pipeline
+        // can avoid handling that complexity and can instead enjoy a simpler, custom built object.
+        // E.G.
+        const entries = result.items
+            .map((item)=> ({
+                title: item.fields.title,
+                description: item.fields.description,
+                weight: item.fields.weight,
+        })).sort((a, b) => b.weight - a.weight);
+
+        dispatch({
+            type: 'Success',
+            payload: { entries },
+        })
+    } catch (error) {
+        dispatch({
+            type: 'Failure',
+            payload: { error },
+        });
+    }
+};
+```
+
+### 3. Conditionally rendering DOM elements in case of preview
+
+Since we're on the topic, you must use a similar strategy in your `App.js` file to conditionally render a DOM element to let the user know they are in a preview state. This must happen once per instance of the App, since we will not perpetuate the `cms-preview` query parameter on route changes.
+
+Class component:
```

Comment:
I would put the hook version first as our boilerplate is no longer using the class version.

Reply:
Moreover, next.js even removed the custom App using class components, so I don’t know if extending is the correct way here.

Reply:
Should we perhaps remove the Class component approach then?

Reply:
Yep I think it’s better.